src/gui/src/renderThread.cpp

```diff
@@ -134,6 +134,8 @@ void RenderThread::run()
     } catch (const std::exception& e) {
       logger_->warn(
           GUI, 102, "An exception occurred during rendering: {}", e.what());
+    } catch (...) {
+      logger_->warn(GUI, 103, "An unknown exception occurred during rendering");
```

Comment:
I would recommend not catching everything and rather add types as they occur. If there  is a completely unknown type of exception, it merits further study and now there is just a silent mystery in the log.

Reply:
You would prefer a stack trace?

Reply:
My preference would be to print the exception, but for the gui to stay function unless it's an error that is critical.

Reply:
critical errors just abort and don't throw an exception.

Reply:
> You would prefer a stack trace?

I would prefer to have a class that semantically defines errors that can be recovered from and to print exception trace for the rest and debug them.

std::bad_alloc for instance, I would not want silenced, I would prefer to exit the app.

Reply:
@QuantamHD @rovinski  any opinion?

Reply:
Agree with @gadfort I think

Reply:
I'm not in favor of crashes so 3-1 against crashing.  In any case it will only come up when the gui is running so a person is likely to see it.